src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,687 @@
+    {"dsVIDEO_ZOOM_PAN_SCAN", (int32_t)dsVIDEO_ZOOM_PAN_SCAN},
+    {"dsVIDEO_ZOOM_LB_2_21_1_ON_4_3", (int32_t)dsVIDEO_ZOOM_LB_2_21_1_ON_4_3},
+    {"dsVIDEO_ZOOM_LB_2_21_1_ON_16_9", (int32_t)dsVIDEO_ZOOM_LB_2_21_1_ON_16_9},
+    {"dsVIDEO_ZOOM_PLATFORM", (int32_t)dsVIDEO_ZOOM_PLATFORM},
+    {"dsVIDEO_ZOOM_16_9_ZOOM", (int32_t)dsVIDEO_ZOOM_16_9_ZOOM},
+    {"dsVIDEO_ZOOM_PILLARBOX_4_3", (int32_t)dsVIDEO_ZOOM_PILLARBOX_4_3},
+    {"dsVIDEO_ZOOM_WIDE_4_3", (int32_t)dsVIDEO_ZOOM_WIDE_4_3},
+    {"dsVIDEO_ZOOM_MAX", (int32_t)dsVIDEO_ZOOM_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsVideoCodingFormat_t
+const static ut_control_keyStringMapping_t dsVideoCodingFormatMappingTable[] = {
+    {"dsVIDEO_CODEC_MPEGHPART2", (int32_t)dsVIDEO_CODEC_MPEGHPART2},
+    {"dsVIDEO_CODEC_MPEG4PART10", (int32_t)dsVIDEO_CODEC_MPEG4PART10},
+    {"dsVIDEO_CODEC_MPEG2", (int32_t)dsVIDEO_CODEC_MPEG2},
+    {"dsVIDEO_CODEC_MAX", (int32_t)dsVIDEO_CODEC_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsVideoCodecHevcProfiles_t
+const static ut_control_keyStringMapping_t dsVideoCodecHevcProfilesMappingTable[] = {
+    {"dsVIDEO_CODEC_HEVC_PROFILE_MAIN", (int32_t)dsVIDEO_CODEC_HEVC_PROFILE_MAIN},
+    {"dsVIDEO_CODEC_HEVC_PROFILE_MAIN10", (int32_t)dsVIDEO_CODEC_HEVC_PROFILE_MAIN10},
+    {"dsVIDEO_CODEC_HEVC_PROFILE_MAINSTILLPICTURE", (int32_t)dsVIDEO_CODEC_HEVC_PROFILE_MAINSTILLPICTURE},
+    {"dsVIDEO_CODEC_HEVC_MAX", (int32_t)dsVIDEO_CODEC_HEVC_MAX},
+    {NULL, -1}
+};
+
+// Mapping table for dsVideoFrameRate_t
+const static ut_control_keyStringMapping_t dsVideoFrameRateTable[] = {
+    {"dsVIDEO_FRAMERATE_UNKNOWN", (int32_t)dsVIDEO_FRAMERATE_UNKNOWN},
+    {"dsVIDEO_FRAMERATE_24", (int32_t)dsVIDEO_FRAMERATE_24},
+    {"dsVIDEO_FRAMERATE_25", (int32_t)dsVIDEO_FRAMERATE_25},
+    {"dsVIDEO_FRAMERATE_30", (int32_t)dsVIDEO_FRAMERATE_30},
+    {"dsVIDEO_FRAMERATE_60", (int32_t)dsVIDEO_FRAMERATE_60},
+    {"dsVIDEO_FRAMERATE_23dot98", (int32_t)dsVIDEO_FRAMERATE_23dot98},
+    {"dsVIDEO_FRAMERATE_29dot97", (int32_t)dsVIDEO_FRAMERATE_29dot97},
+    {"dsVIDEO_FRAMERATE_50", (int32_t)dsVIDEO_FRAMERATE_50},
+    {"dsVIDEO_FRAMERATE_59dot94", (int32_t)dsVIDEO_FRAMERATE_59dot94},
+    {"dsVIDEO_FRAMERATE_MAX", (int32_t)dsVIDEO_FRAMERATE_MAX},
+    {NULL, -1}
+};
+
+/**
+ * @brief This function clears the stdin buffer.
+ *
+ * This function clears the stdin buffer.
+ */
+void readAndDiscardRestOfLine(FILE* in)
+{
+    int c;
+    while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/*FrameratePreChange callback*/
+void dsVideoDevice_FrameratePreChange(unsigned int tSecond)
+{
+    UT_LOG_INFO("FrameratePreChange callback tSecond:[%d]",tSecond);
+}
+
+/*FrameratePostChange callback*/
+void dsVideoDevice_FrameratePostChange(unsigned int tSecond)
+{
+    UT_LOG_INFO("FrameratePostChange callback tSecond:[%d]",tSecond);
+}
+
+static void dsVideoDevice_getHandle()
+{
+    dsError_t status   = dsERR_NONE;
+    int32_t device, selectedVideoDevice;
```

Comment:
initialize device and selectedVideoDevice variables.

Reply:
done